### DISCUSSION TITLE
org.eclipse.m2e:lifecycle-mapping is reported as a warning for maven

### DISCUSSION THREAD

**Discussion by hazendaz** [renovatebot/renovate#22579] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

org.eclipse.m2e:lifecycle-mapping is not a real plugin with maven.  Its a placeholder for m2e (maven to eclipse) configuration.  Currently renovate complains about this as follows.

⚠ Dependency Lookup Warnings ⚠

    Renovate failed to look up the following dependencies: Failed to look up maven package org.eclipse.m2e:lifecycle-mapping.

Can this one be added to an ignores list as it won't ever exist?

Details about m2e can be found here https://www.eclipse.org/m2e/documentation/m2e-execution-not-covered.html.  Not all maven projects have this but those that do, renovate has always produced a warning as it cannot find the artifact since there isn't actually one to to find.

### Relevant debug logs

all details on described problem.  Example project renovate issue tracker: https://github.com/hazendaz/base-parent/issues/437

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by HonkingGoose** [comment]
Did you try disabling Renovate for the placeholder package?

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "packageRules": [
    {
      "matchPackageNames": ["put-the-correct-m2e-package-name-here"],
      "enabled": false
    }
  ]
}
```

Also, can you explain _why_ you have a placeholder m2e package in the first place? Maybe the problem is with the package file?

**Reply by hazendaz** [reply]
I have not but think its widespread usage that it should be built into renovate itself as it won't ever resolve to anything and its used heavily by projects that are worked in Eclipse IDE where m2e processes.  I've been ignoring it for few years now from memory but thought to open this so it possibly just gets fixed in Renovate so it doesn't show up as a warning as there is nothing to warn about.  Preference there not having to setup unnecessary setup just because across maven repos.  This is the only such item in maven I'm aware of like this.  Its been a problem for every package manager I can think of.  mvnrepository even added a dummy entry for it but maven central has no such dummy entry.

**Reply by hazendaz** [reply]
I should add, its verbatim what I noted there.  There is only the one.  Its a simple marker so that configuration and be applied to m2e where plugins failed to support it and/or to override if needed.  It won't be anything but that single entry for all.

**Comment by rarkins** [comment]
@Churro are you familiar with this package?

**Reply by Churro** [reply]
To some extent, yes. It is correct that `org.eclipse.m2e:lifecycle-mapping` is a placeholder package that was never published to Maven Central. 

To get rid of the warning, you can either add a remote repository with a dummy (like [this](https://simulation.tudelft.nl/maven/org/eclipse/m2e/lifecycle-mapping/)) or ignore the package as mentioned above. The smartest way would be to update the `pom.xml` to [a more recent syntax](https://www.eclipse.org/m2e/documentation/release-notes-17.html#new-syntax-for-specifying-lifecycle-mapping-metadata) for lifecycle mapping metadata that no longer requires the fake package.

GitHub has [indexed](https://github.com/search?q=%22%3CartifactId%3Elifecycle-mapping%3C%2FartifactId%3E%22+NOT+is%3Afork+path%3Apom.xml&type=code&p=1) at least 27k repos (without forks), where this occurs. I think it would make sense to add a general workaround that disables `org.eclipse.m2e:lifecycle-mapping`. Should I open a PR?

**Reply by hazendaz** [reply]
@Churro You have me thinking that maybe the right answer is to talk to the source of the situation.  I've raised this https://github.com/eclipse-m2e/m2e-core/issues/1412.  Let's see what they say.  Seems reasonable they could load a dummy jar and even have a bit of substance in it to highlight all that can be done with m2e which would better resolve this.

If that is a no go, then yes I think this is needed.  The more recent syntax you noted, seems odd, never seen it used myself, never even noticed that after using maven for 10 years and heavy Eclipse user for 14 years.  Just doesn't jump out in release notes for 1.7.0 of m2e but does in that demo but it lacks showing all variations and check in m2e, same.  They note it can be at plugin level but its not there and they have not deprecated the old way either.  I think even at 1.7.0 it noted there was no quick fix and I don't seem to recall say in last 2 years that there was any quick fix yet either that didn't setup the way it is.

**Reply by Churro** [reply, marked as answer]
@hazendaz, I have found [Bug 367870](https://bugs.eclipse.org/bugs/show_bug.cgi?id=367870), which discussed the same issue in 2012 already, without positive outcome. Hence, PR #22671 adds a workaround.

**Comment by hazendaz** [comment]
This looks great thank you.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
